### PR TITLE
HPXML Integration

### PR DESCRIPTION
**Pull request by nmerket**
#### What's this PR do?

This provides an HPXML import/export capability through the API similar to the BuildingSync capability already on this branch.

#### How should this be manually tested?

Write a script to do some API calls. I have one started if you're interested.

#### What are the relevant tickets?

I didn't make a ticket.

#### Definition of Done:
- [x] Is there appropriate test coverage? (e.g. ChefSpec, Mocha/Chai, Python, etc.)
- [x] Does this PR require a Selenium test? (e.g. Browser-specific bugs or complicated UI bugs)
- [x] Does this PR require a regression test? All fixes require a regression test.
- [x] Does this add new dependencies? If so, does PIP, npm, bower requirements need to be updated?
**Yes, run pip install**

### CODE REVIEW

**Comment by nmerket** [comment]
@nllong It looks like the failing tests are coming from cached environments. Can you :boom: those so it will reload?

**Comment by nllong** [comment]
done and restarted.

File: seed/hpxml/tests/test_hpxml.py [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+from seed.lib.superperms.orgs.models import (
+    Organization,
+    OrganizationUser,
+)
+from seed.test_helpers.fake import (
+    FakePropertyStateFactory,
+)
+
+
+class TestHPXML(TestCase):
```
Comment (nllong) [inline seed/hpxml/tests/test_hpxml.py]: thanks for adding a bunch of tests!

File: seed/migrations/0078_hpxml_building_file.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+    dependencies = [
+        ('seed', '0077_auto_20171024_1213'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='buildingfile',
+            name='file_type',
+            field=models.IntegerField(choices=[(0, 'Unknown'), (1, 'BuildingSync'), (2, 'GeoJSON'), (3, 'HPXML')], default=0),
```
Comment (nllong) [inline seed/migrations/0078_hpxml_building_file.py]: i will probably merge this migration into the larger bricr migration once I merge this down. just FYI

File: seed/models/building_file.py [hunk 3]
```diff
@@ -140,7 +157,7 @@ def process(self, organization_id, cycle, property_state=None, property_id=None)
         self.save()
 
         # add in the measures
-        for m in data['measures']:
+        for m in data.get('measures', []):
```
Comment (nllong) [inline seed/models/building_file.py]: 👍

File: seed/hpxml/hpxml.py [hunk 4]
```diff
@@ -0,0 +1,419 @@
...
+class HPXMLError(Exception):
+    pass
+
+
+class HPXML(object):
+    NS = 'http://hpxmlonline.com/2014/6'
+
+    HPXML_STRUCT = {
+        'address_line_1': {
+            'path': 'h:Site/h:Address/h:Address1',
```
[the remark this answers is not in the document]
Comment (nmerket) [reply seed/hpxml/hpxml.py]: It's shorthand for the HPXML namespace. It's used [here](https://github.com/SEED-platform/seed/blob/0f4ac74fe03aa23a3b6ba2ec842dd3ad59728f08/seed/hpxml/hpxml.py#L33) and [here](https://github.com/SEED-platform/seed/blob/0f4ac74fe03aa23a3b6ba2ec842dd3ad59728f08/seed/hpxml/hpxml.py#L90).

**Pull request merged**